hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -289,6 +293,84 @@ private void sendTaskFailed(ExecutorDriver driver, Task task) {
       .setState(TaskState.TASK_FAILED)
       .build());
   }
+  
+  protected void runHealthChecks(ExecutorDriver driver, Task task) {
+    String taskIdStr = task.getTaskInfo().getTaskId().getValue();
+    log.info("Performing health check for task: " + taskIdStr);
+
+    int healthCheckPort = getHealthCheckPort(taskIdStr);
+    boolean taskHealthy = false;
+
+    if (healthCheckPort != -1) {
+      Socket socket = null;
+      try {
+        // TODO (elingg) with better process supervision, check which process is
+        // bound to the port.
+        // Also, possibly do a http check for the name node UI as an additional
+        // health check.
+        String localhostAddress = InetAddress.getLocalHost().getHostAddress();
+        socket = new Socket();
+        socket.bind(new InetSocketAddress(localhostAddress, healthCheckPort));
+      } catch (IOException e) {
+        taskHealthy = true;
+        log.info("Could not bind to port " + healthCheckPort + ", port is in use as expected.");
+      } catch (SecurityException | IllegalArgumentException e) {
+        log.error("Exception determining if a port is in use: ", e);
+      } finally {
+        if (socket != null) {
+          try {
+            socket.close();
+          } catch (IOException e) {
+            log.error("Error closing socket in health check: ", e);
+          }
+        }
+      }
+    }
+    
+    if (!taskHealthy) {
+      log.fatal("Node health check failed for task: " + taskIdStr);
+      killTask(driver, task.getTaskInfo().getTaskId());
```

Comment:
Yes, we want to. And this is why we actually terminate the executor which will kill both tasks. I have a note here that with better process supervision, we should not have to kill the full executor, but I think there's work to be done there including monitoring the ZK task in our persistent and live state and I'd rather play it safe in the meantime.